scipy/_lib/_array_api.py

```diff
@@ -545,3 +551,52 @@ def xp_sign(x: Array, /, *, xp: ModuleType | None = None) -> Array:
     sign = xp.where(x < 0, -one, sign)
     sign = xp.where(x == 0, 0*one, sign)
     return sign
+
+
+_SCIPY_ARRAY_API = os.environ.get("SCIPY_ARRAY_API", False)
+array_api_compat_prefix = "scipy._lib.array_api_compat"
+
+
+def get_array_subpackage_func(f_name, xp, n_array_args, root_namespace,
+                              subpackage, generic_implementations):
```

Comment:
When generalizing the signature, I didn't think terribly carefully about the new argument names, whether they should be keyword only, etc.
- `root_namespace` - the name of the private module in which SciPy's implementation lies. For instance, for the `special` functions, it's `_ufunc`. Some generalization may be needed if the SciPy implementations are spread across multiple files. There are lots of possible solutions. The most obvious is to call `get_array_subpackage_func` with the appropriate `root_namespace` for the given `f_name`; another possibility is to import all the SciPy implementations into a single, private module, and set `root_namespace` to the name of that module.
- `subpackage` - e.g. 'signal', 'linalg', 'special'
- `generic_implementations` - a dictionary of callables that defines functions that may not have direct equivalents in all other backends but can be defined in terms of other `xp` or `spx` functions. We use this in `special` to define `xlogy` as `x * xp.log(y)` and `chdtrc` in terms of `gammaincc` (when available), for instance.

The intent was to produce a proof of concept; we can wordsmith everything to our heart's content during review.